code/game/objects/items.dm

```diff
@@ -1066,3 +1066,65 @@ var/global/list/image/blood_overlays = list()
 
 /obj/item/animationBolt(var/mob/firer)
 	new /mob/living/simple_animal/hostile/mimic/copy(loc, src, firer, duration=SPELL_ANIMATION_TTL)
+
+/obj/item/proc/is_worn(mob/user)
+	var/mob/living/carbon/monkey/M
+	var/mob/living/carbon/human/H
+	if(istype(user, /mob/living/carbon/monkey))
+		M = user
+	if(istype(user, /mob/living/carbon/human))
+		H = user
+	if(!(H || M))
+		return 0
+	if(H)
+		if(slot_flags & SLOT_OCLOTHING)
```

Comment:
Oh god.

It's probably better to iterate over the active bitflags of the `slot_flags` field and then do `H.get_item_by_slot(slot)`

Probably something along the lines of the following:
```dm
for (var/bit = 0 to 15)
    bit = 1 << bit
    if (bit & slot_flags && H.get_slot_by_id(bit))
        return TRUE
```